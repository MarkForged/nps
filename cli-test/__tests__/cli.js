import {resolve} from 'path'
import runNPS from './helpers/run-nps'

const fixturesPath = resolve(__dirname, './fixtures')

test('with config with default script', () =>
  snapshot('-c ./package-scripts-with-default.js'))

test('with a missing config', () =>
  snapshot('-c ./something-that-does-not-exist.js'))

test('with silent option set true', () =>
  snapshot('-c ./package-script-silent.js'))

test('with --require', () =>
  snapshot('--config ./es6-package-scripts.js --require babel-register log'))

test('with --get-yargs-completions', () =>
  snapshot('--config ./package-scripts.js --get-yargs-completions li'))

test('with prefix', () => snapshot('--config ./package-scripts.js lint.s.t.s'))

test('with --no-scripts', () => snapshot('test --no-scripts'))

function snapshot(args) {
  return runNPS(fixturesPath, args).then(results => {
    const snapshottableResults = convertResultToLinuxSpecific(results)
    expect(snapshottableResults).toMatchSnapshot()
  })
}

function convertResultToLinuxSpecific(results) {
  return removeUnwantedQuotes(relativizePaths(results))
}

/**
 * This takes the results object and removes environment-specific
 * elements from the path.
 * @param {Object} results - This is the results object from runNPS
 * @return {Object} - The new results object with the clean paths
 */
function relativizePaths(results) {
  return Object.keys(results).reduce((obj, key) => {
    obj[key] = results[key]
      .replace(':/', ':\\')
      .replace(resolve(__dirname, '../'), '<projectRootDir>')
      .replace(/\\/g, '/')
    return obj
  }, {})
}

/**
 * This takes the results object and removes unwanted quotes
 * @param {Object} results - This is the results object from runNPS
 * @return {Object} - The new results object without unwanted quotes
 */
function removeUnwantedQuotes(results) {
  const splittedStdout = results.stdout.split(/\r?\n/)
  const {length} = splittedStdout
  if (length > 1) {
    splittedStdout[length - 2] = splittedStdout[length - 2].replace(/"/g, '')
  }
  const joinedStdoutResult = splittedStdout.join('\n')
  return Object.assign(results, {
    stdout: joinedStdoutResult,
  })
}
