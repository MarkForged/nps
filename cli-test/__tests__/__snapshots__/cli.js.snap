// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with --get-yargs-completions 1`] = `
Object {
  "stderr": "",
  "stdout": "like
lint
lint.sub.thing
lint.sub.hiddenThing
",
}
`;

exports[`with --no-scripts 1`] = `
Object {
  "stderr": "",
  "stdout": "nps is executing \`test\`
test script
",
}
`;

exports[`with --require 1`] = `
Object {
  "stderr": "",
  "stdout": "nps is executing \`log\` : echo \\"log\\"
log
",
}
`;

exports[`with silent option set true 1`] = `
Object {
  "stderr": "",
  "stdout": "silent
",
}
`;

exports[`with a missing config 1`] = `
Object {
  "stderr": "Unable to find JS config at \\"./something-that-does-not-exist.js\\". https://github.com/kentcdodds/nps/blob/v0.0.0-semantically-released/other/ERRORS_AND_WARNINGS.md#unable-to-find-config
",
  "stdout": "",
}
`;

exports[`with config with default script 1`] = `
Object {
  "stderr": "",
  "stdout": "nps is executing \`default\` : echo \\"default script\\"
default script
",
}
`;

exports[`with prefix 1`] = `
Object {
  "stderr": "",
  "stdout": "nps is executing \`lint.sub.thing.script\` : echo \\"deeply nested thing\\"
deeply nested thing
",
}
`;
